src/test/optimizer/strategy/or_to_union_rule_test.cpp

```diff
@@ -0,0 +1,223 @@
+#include "gtest/gtest.h"
+
+#include "logical_query_plan/union_node.hpp"
+#include "optimizer/strategy/or_to_union_rule.hpp"
+#include "strategy_base_test.hpp"
+#include "testing_assert.hpp"
+
+namespace opossum {
+
+class OrToUnionRuleTest : public StrategyBaseTest {
+ public:
+  void SetUp() override {
+    node_a = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}, {DataType::Int, "c"}}, "a");
+    a_a = node_a->get_column("a");
+    a_b = node_a->get_column("b");
+
+    node_b = MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}}, "b");
+    b_a = node_b->get_column("a");
+    b_b = node_b->get_column("b");
+
+    node_c = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}, {DataType::Int, "c"}}, "c");
+    c_a = node_c->get_column("a");
+    c_b = node_c->get_column("b");
+
+    customer_demographics =
+        MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "c_customer_sk"}}, "customer_demographics");
```

Comment:
Instead of OptimizeTPCDS35, how about testing an LQP where the PredicateNode has multiple outputs?

Reply:
Do you mean a diamond structure? Good idea, this would be an interesting test case.

Reply:
I added a test for it.